cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -296,6 +296,215 @@ CAST_NUMERIC_FROM_VARBINARY(double, arrow::DoubleType, FLOAT8)
+    valid_entry = -1;
+    return -1;
+  }
+
+  if (radix <= 10) {
+    if (value >= '0' && value < '0' + radix) {
+      return value - '0';
+    } else {
+      valid_entry = -1;
+      return -1;
+    }
+  } else if (value >= '0' && value <= '9') {
+    return value - '0';
+  } else if (value >= 'a' && value < 'a' + radix - 10) {
+    return value - 'a' + 10;
+  } else if (value >= 'A' && value < 'A' + radix - 10) {
+    return value - 'A' + 10;
+  }
+  valid_entry = -1;
+  return -1;
+}
+
+GDV_FORCE_INLINE
+void byte2char(int32_t radix, int32_t fromPos, char* value, int32_t valueLen) {
+  for (int i = fromPos; i < valueLen; i++) {
+    value[i] = static_cast<char>(character_for_digit(value[i], radix));
+  }
+}
+
+GDV_FORCE_INLINE
+void char2byte(int32_t radix, int32_t fromPos, char* value, int32_t valueLen,
+               int32_t* valid_entry) {
+  for (int i = fromPos; i < valueLen; i++) {
+    value[i] = static_cast<char>(character_digit(value[i], radix, *valid_entry));
+    if (*valid_entry != 1) {
+      break;
+    }
+  }
+}
+
+GANDIVA_EXPORT
+const char* conv_int64_int32_int32(int64_t context, int64_t in, int32_t from_base,
+                                   int32_t to_base, int32_t* out_len) {
+  std::string to_utf8 = std::to_string(in);
+  char* in_utf8 = &to_utf8[0];
+  auto in_utf8_len = static_cast<int32_t>(to_utf8.length());
+
+  return conv_utf8_int32_int32(context, in_utf8, in_utf8_len, from_base, to_base,
+                               out_len);
+}
+
+GANDIVA_EXPORT
+const char* conv_int32_int32_int32(int64_t context, int32_t in, int32_t from_base,
+                                   int32_t to_base, int32_t* out_len) {
+  std::string to_utf8 = std::to_string(in);
+  char* in_utf8 = &to_utf8[0];
+  auto in_utf8_len = static_cast<int32_t>(to_utf8.length());
+
+  return conv_utf8_int32_int32(context, in_utf8, in_utf8_len, from_base, to_base,
+                               out_len);
+}
+
+GANDIVA_EXPORT
+const char* conv_utf8_int32_int32(int64_t context, const char* in, int32_t in_len,
+                                  int32_t from_base, int32_t to_base, int32_t* out_len) {
+  if (in_len <= 0) {
+    out_len = 0;
+    return "";
+  }
+
+  int32_t valueLen = 64;
+  char* value = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, valueLen));
+  char* num = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, in_len));
+
+  if (value == nullptr || num == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    out_len = 0;
+    return "";
+  }
+
+  int fromBs = from_base;
+  int toBs = to_base;
+
+  if (fromBs < -36 || fromBs > 36 || fromBs == 0 || fromBs == 1 || abs(toBs) < -36 ||
+      abs(toBs) > 36 || abs(toBs) == 0 || abs(toBs) == 1) {
+    // Checking if the variable is in range limit
+    gdv_fn_context_set_error_msg(context,
+                                 "The numerical limit of this variable is out range");
+    *out_len = 0;
+    return "";
+  }
```

Comment:
Why is this check required? Also, remove redundant conditionals like abs(toBs) < -36.

Reply:
This is not redundant, using the value numeric_limits<int>::min() like toBs, this condition (**abs(toBs) < -36**) is true
I believe we will need to keep this validation

this check is necessary as we will only handle base inputs and outputs between -36 and +36 with the exception of 1 and 0, as specified in Hive function